src/libfetchers/git.cc

```diff
@@ -392,6 +401,28 @@ struct GitInputScheme : InputScheme
         AutoDelete delTmpDir(tmpDir, true);
         PathFilter filter = defaultPathFilter;
 
+        RunOptions checkCommitOpts(
+            "git",
+            { "-C", repoDir, "cat-file", "commit", input.getRev()->gitRev() }
+        );
+        checkCommitOpts.searchPath = true;
+        checkCommitOpts.mergeStderrToStdout = true;
+
+        auto result = runProgram(checkCommitOpts);
+        if (WEXITSTATUS(result.first) == 128
+            && result.second.find("bad file") != std::string::npos
```

Comment:
Checking for strings like "bad file" in stderr seems pretty fragile... Do we actually need this test? Presumably a wrong `rev` will fail in the subsequent call to `git archive`.

Reply:
I'm afraid no: if `git cat-file` errors, the exit code is always 128, the reason for the error is on stderr.